From bd20de50347becb6fdf7faf72c0bcb3dcba31c2c Mon Sep 17 00:00:00 2001
From: Thomas Richter <tmricht@linux.vnet.ibm.com>
Date: Mon, 6 Aug 2012 15:20:29 -0700
Subject: [PATCH] lldpad: remove unreferenced functions

Remove functions remove_all_bond_ports(), remove_bond_port(), add_bond_port(),
recv_on_bond() functions which are not referenced anymore after the bond
support has been added.

Signed-off-by: Thomas Richter <tmricht@linux.vnet.ibm.com>
Signed-off-by: John Fastabend <john.r.fastabend@intel.com>
---
 lldp/l2_packet.h       |    5 --
 lldp/l2_packet_linux.c |  118 ------------------------------------------------
 lldpad.c               |    2 -
 3 files changed, 125 deletions(-)

diff --git a/lldp/l2_packet.h b/lldp/l2_packet.h
index dabfccd..69b45c0 100644
--- a/lldp/l2_packet.h
+++ b/lldp/l2_packet.h
@@ -144,9 +144,4 @@ int l2_packet_send(struct l2_packet_data *l2, const u8 *dst_addr, u16 proto,
 
 void l2_packet_get_port_state(struct l2_packet_data *, u8 *);
 
-struct port * add_bond_port(const char *ifname);
-int remove_bond_port(const char *ifname);
-void recv_on_bond(void *ctx, int ifindex, const u8 *buf, size_t len);
-void remove_all_bond_ports(void);
-
 #endif /* L2_PACKET_H */
diff --git a/lldp/l2_packet_linux.c b/lldp/l2_packet_linux.c
index 1ffdd92..f7d6d95 100644
--- a/lldp/l2_packet_linux.c
+++ b/lldp/l2_packet_linux.c
@@ -42,8 +42,6 @@
 #include "lldp/states.h"
 #include "lldp_dcbx_nl.h"
 
-static struct port *bond_porthead = NULL;
-
 struct l2_packet_data {
 	int fd;
 	char ifname[IFNAMSIZ + 1];
@@ -310,119 +308,3 @@ void l2_packet_get_port_state(struct l2_packet_data *l2, u8  *portEnabled)
 	close(s);
 	return;
 }
-
-
-struct port *add_bond_port(const char *ifname)
-{
-	struct port *bond_newport;
-
-	bond_newport = bond_porthead;
-	while (bond_newport != NULL) {
-		if(!strncmp(ifname, bond_newport->ifname,
-			MAX_DEVICE_NAME_LEN))
-			return bond_newport;
-		bond_newport = bond_newport->next;
-	}
-
-	bond_newport  = (struct port *)malloc(sizeof(struct port));
-	if (bond_newport == NULL) {
-		syslog(LOG_ERR, "failed to malloc bond port %s", ifname);
-		return NULL;
-	}
-	memset(bond_newport,0,sizeof(struct port));	
-	bond_newport->next = NULL;
-	bond_newport->ifname = strdup(ifname);
-	if (bond_newport->ifname == NULL) {
-		syslog(LOG_ERR, "failed to strdup bond name %s", ifname);
-		goto fail1;
-	}
-	
-	bond_newport->l2 = l2_packet_init(bond_newport->ifname, NULL,
-		ETH_P_LLDP, recv_on_bond, bond_newport, 1);
-
-	if (bond_newport->l2 == NULL) {
-		syslog(LOG_ERR, "failed to open register layer 2 access to "
-			   "ETH_P_LLDP");
-		goto fail2;
-	}
-
-	if (bond_porthead)
-		bond_newport->next = bond_porthead;
-	bond_porthead = bond_newport;
-
-	return bond_newport;
-
-fail2:
-	free(bond_newport->ifname);
-	bond_newport->ifname = NULL;
-fail1:
-	free(bond_newport);
-	bond_newport = NULL;
-	return NULL;
-}
-
-
-void recv_on_bond(UNUSED void *ctx, int ifindex, const u8 *buf, size_t len)
-{
-	struct port *port;
-
-	/* Find the originating slave port object */
-	for (port = porthead; port != NULL && port->l2->ifindex != ifindex;
-		port = port->next)
-		;
-
-	if (port)
-		rxReceiveFrame(port, ifindex, buf, len);
-}
-
-
-int remove_bond_port(const char *ifname)
-{
-	struct port *bond_port = bond_porthead;
-	struct port *bond_parent = NULL;
-
-	while (bond_port != NULL) {
-		if (!strncmp(ifname, bond_port->ifname, MAX_DEVICE_NAME_LEN)) {
-			LLDPAD_DBG("In remove_bond_port: Found bond port  %s\n",
-				bond_port->ifname);
-			break;
-		}
-		bond_parent = bond_port;
-		bond_port = bond_port->next;
-	}
-
-	if (bond_port == NULL)
-		return -1;
-
-	l2_packet_deinit(bond_port->l2);
-
-	if (bond_parent == NULL)
-		bond_porthead = bond_port->next;
-	else if (bond_parent->next == bond_port)     /* sanity check */
-		bond_parent->next = bond_port->next;	
-	else
-		return -1;
-
-	if (bond_port->ifname)
-		free(bond_port->ifname);
-	
-	free(bond_port);
-	return 0;
-}
-
-void remove_all_bond_ports(void)
-{
-	struct port *bond_port = bond_porthead;
-	struct port *next_bond_port = bond_porthead;
-
-	while (bond_port != NULL) {
-		next_bond_port = bond_port->next;
-
-		l2_packet_deinit(bond_port->l2);
-		if (bond_port->ifname)
-			free(bond_port->ifname);
-		free(bond_port);
-
-		bond_port = next_bond_port;
-	}
-}
diff --git a/lldpad.c b/lldpad.c
index cd46766..ca1f7c5 100644
--- a/lldpad.c
+++ b/lldpad.c
@@ -168,7 +168,6 @@ lldpad_reconfig(UNUSED int sig, UNUSED void *eloop_ctx, UNUSED void *signal_ctx)
 	clean_lldp_agents();
 	deinit_modules();
 	remove_all_adapters();
-	remove_all_bond_ports();
 	destroy_cfg();
 
 	/* Reinit config file and modules */
@@ -424,7 +423,6 @@ int main(int argc, char *argv[])
 	clean_lldp_agents();
 	deinit_modules();
 	remove_all_adapters();
-	remove_all_bond_ports();
 	ctrl_iface_deinit(clifd);  /* free's clifd */
 	event_iface_deinit();
 	stop_lldp_agents();
-- 
1.7.10.4

